crates/router/src/services/api/client.rs

```diff
@@ -152,20 +158,31 @@ impl ProxyClient {
         base_url: String,
         client_certificate: Option<String>,
         client_certificate_key: Option<String>,
-    ) -> reqwest::Client {
-        // Fix this shit as well
+    ) -> CustomResult<reqwest::Client, ApiClientError> {
         match (client_certificate, client_certificate_key) {
-            (None, None) => {
+            (Some(certificate), Some(certificate_key)) => {
+                let client_builder =
+                    reqwest::Client::builder().redirect(reqwest::redirect::Policy::none());
+                let identity = payments::helpers::create_identity_from_certificate_and_key(
+                    certificate,
+                    certificate_key,
+                )?;
+                Ok(client_builder
+                    .identity(identity)
+                    .build()
+                    .into_report()
+                    .change_context(ApiClientError::ClientConstructionFailed)
+                    .attach_printable(
+                        "Failed to construct client with certificate and certificate key",
+                    )?)
+            }
+            (_, _) => {
                 if self.whitelisted_urls.contains(&base_url) {
-                    self.non_proxy_client.clone()
+                    Ok(self.non_proxy_client.clone())
                 } else {
-                    self.proxy_client.clone()
+                    Ok(self.proxy_client.clone())
```

Comment:
if a  client certificate is present we should build a proxied client (unless whitelisted URL's)
you can store the proxy config in the `ProxyClient` struct for this case